### ISSUE TITLE
Widget overlap

### ISSUE THREAD

**Issue by cdupont** [jtdaugherty/brick#386]
I have two widgets of the same size (say, 5*5). Can I overlap them?
The bottom one would "show" through the empty spaces of the top one.

**Comment by jtdaugherty** [comment]
It's possible to do that with layers. That is mentioned in the User Guide [here](https://github.com/jtdaugherty/brick/blob/master/docs/guide.rst#appdraw-drawing-an-interface). To make it work, elements of the top layer need to be translated into place with `Brick.Widgets.Core.translateBy`. If you want to share a bit more about what you are trying to accomplish, I'm happy to provide thoughts and ideas on how best to go about it.

**Comment by cdupont** [comment]
I was thinking to do a similar process of layers, but for a local area only. Something like `overlay :: [Widget] -> Widget`. 
My use case is that I have various tiles in my app, and sometime I need to merge two tiles. So the features of the two tiles need to be merged and appear on only one tile. Maybe better to do it by hand anyway :)

**Comment by jtdaugherty** [comment]
Okay, I see, thanks. Unfortunately there is not a way to do layers on a local basis only; they can only be done globally and translated to the desired location. There is an API for doing that easily, but I suspect that given what you've said about what you want to accomplish, it would be best to decide on a tile by tile basis what should be displayed rather than using layers. I also suggest you consider using a single `Widget` with a custom Vty image build process, rather than a grid of `Widget`s, to build your tile display. There will likely be a performance concern if you use a `Widget` for each tile due to the cost of box layout. When you get to that point, I'm happy to help if you run into any issues!

**Comment by cdupont** [comment]
Thanks.
Anyway, is there a way to just replace a character in a given Widget? I'm using a widget that includes a border. However I need to replace one of the segment of the border with another character.
```
┌─────┐                                                                                   
│     │
→      │
│     │
└─────┘
```

**Comment by jtdaugherty** [comment]
If you need to do that, I recommend using the border-drawing functions in `Brick.Widgets.Border` to build your own border (or not, as you see fit). There is not a way to selectively modify the content of a `Widget`.

**Comment by jtdaugherty** [comment]
As a reference, you could use [the `border` implementation](https://hackage.haskell.org/package/brick-1.1/docs/src/Brick.Widgets.Border.html#border) as a starting point.

**Comment by yhuangsh** [comment]
I was trying to draw this: a queue of cards lined up from left to right with each one overlapping half of the one before it. I first tried to add negative padding, then realized padding can only be positive. Then I tried to use translationBy using negative x-coordinate. It worked, only I found the negatively offset cards were rendered under the one before it, which is the exact opposite I wanted. I ended up using cropping. It worked perfectly except if I want to pop some of the cards up from the queue, I will have to patch the parts of popped card(s) missing from the cropping, which becomes quite messy. 

I will be thrilled to learn the right way to achieve this. 

My observation: it's a bit counter-intuitive that one lays the widgets out from left to right , top to bottom, but the rendering (or the outcome of the rendering when there is any overlap) isn't following the same order. Not a bug for sure and I'm only grateful for this library for what it allows me to do.
